site/_posts/2023-07-11-get-more-value-out-of-fusionauth.md

```diff
@@ -0,0 +1,313 @@
+        {
+          LogResults($"User creation failed. {userCreatedReturnValue.result}");
+        }
+      }
+      else
+      {
+        LogResults($"User data provided is invalid");
+      }
+    }
+```
+
+### Creating the User
+
+Because you decorated your User objects correctly using the JSONProperty attribute, you can use the JSON Serializer to serialize the object with the `JsonConvert.SerializeObject` method. You also need to encode it properly before you submit it. Once that is complete, you can submit the request to the FusionAuth server. You will do so here using the async pattern so as to not hold up the processing for a request that may take some time or give the application an unresponsive feel. Once you receive the return values, you can interrogate them and act accordingly.
+
+{% comment %}
+{% convert code to RemoteInclude with tags for this section when converting to Astro %}
+{% RemoteCode url="https://raw.githubusercontent.com/FusionAuth/fusionauth-example-dotnet-windowsform-api/main/sourcecode/FusionDemoAPI/FusionAuthClient.cs" lang="c#" tags="CreateUser" %}
+{% endcomment %}
+```cs
+  public async Task<ReturnValue> CreateUser(User userToCreate)
+    {
+      //returns HttpClient with correct setting for instance
+      HttpClient httpClient = GetClient();
+
+      //initialize the return value 
+      ReturnValue returnVal = new ReturnValue()
+      {
+        success = false
+      };
+
+      try
+      {
+        //set the endpoint for creating a user
+        string apiURL = $"/api/user";
+
+        var payload = JsonConvert.SerializeObject(userToCreate);
+        var content = new StringContent(payload, Encoding.UTF8, "application/json");
+
+        HttpResponseMessage response = await httpClient.PostAsync($"{FUSIONAUTH_HOST}{apiURL}", content);
+
+        if (response.IsSuccessStatusCode)
+        {
+
+          string responseBody = await response.Content.ReadAsStringAsync();
+          JObject result = JObject.Parse(responseBody);
+          returnVal.success = true;
+          //assign the newly created user id to the result
+          returnVal.result = result["user"]["id"].ToString();
+        }
+
+      }
+      catch (Exception ex)
+      {
+        Console.WriteLine($"There was an error in the request {ex.Message}");
+        returnVal.result = ex.Message;
+      }
+
+      return returnVal;
+    }
+```
+
+## User Workflow with FusionAuth APIs
+*“Amateurs do it till they get it right. Professionals do it until they can not get it wrong.”*
+
+You may have originally set up the users manually in the FusionAuth admin user interface and that is fine to get started. However, as you add more steps and requirements to creating a user, the more likely something will be missed. There can be several steps required to set up a user correctly such as making sure they belong to the correct group(s). While you may be infallible as a developer and get it right every time, wink wink, most likely your users will not.
```

Comment:
```suggestion
You may have originally set up the users manually in the FusionAuth admin user interface and that is fine to get started. However, the more steps and requirements you have for creating a user, the more likely something will be missed. There can be several steps required to set up a user correctly such as making sure they belong to the correct group(s). While you may be infallible as a developer and get it right every time, wink wink, most likely your users will not.
```

I have found it helpful to read a post aloud to see if there are any awkward turns of phrase.